lib/promises.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/// encapsulate a method with a node-style callback in a Promise
+/// @param {object} 'this' of the encapsulated function
+/// @param {function} function to be encapsulated
+/// @param {Array-like} args to be passed to the called function
+/// @return {Promise} a Promise encapuslaing the function
+module.exports.promise = function (fn, context, args) {
+
+    //polyfill (for 0.12)
+    //avoid [].slice.call due to performance issues on v8
```

Comment:
The code uses `Array.from` if available. But that is not available in node 0.12 so `[].slice.call` is used instead. `args` in most cases will be an `arguments` abject passed from another method.